crates/forge/bin/cmd/soldeer.rs

```diff
@@ -0,0 +1,105 @@
+use clap::Parser;
+use eyre::Result;
+
+extern crate soldeer_lib;
+
+use soldeer_lib::commands::{Args, Install, Login, Push, Subcommands, Update, VersionDryRun};
+use yansi::Paint;
+
+// CLI arguments for `forge soldeer`.
+#[derive(Clone, Debug, Parser)]
+#[clap(override_usage = "forge soldeer install [DEPENDENCY]~[VERSION] <REMOTE_URL>
+    forge soldeer push [DEPENDENCY]~[VERSION] <CUSTOM_PATH_OF_FILES>
+    forge soldeer login
+    forge soldeer update
+    forge soldeer version")]
+pub struct SoldeerArgs {
+    /// Command must be one of the following install/push/login/update/version.
+    #[clap(required = true)]
+    command: String,
+
+    /// Only push and install can have arguments.
+    #[clap(required = false)]
+    args: Option<Vec<String>>,
+}
```

Comment:
Can we not re-use Clap implementation from soldeer? Why are we manually parsing arguments?

Reply:
Those arguments are like parameters that needs to be sent to the lib. I used them to not rewrite the lib with custom params

Reply:
do you have a recommendation on how to approach this?

Reply:
Well I see you have a clap [`Subcommands`](https://github.com/mario-eth/soldeer/blob/051f70cfe811efc3563babeeee6f6eff17b168e6/src/commands.rs#L19), you should be able to do `Soldeer(#[command(subcommand)] soldeer::Subcommands)` in the forge enum

Reply:
Solved. Thanks for the tip.